src/app/deeds/deeds.component.ts

```diff
@@ -79,24 +96,22 @@ export class DeedsComponent implements OnInit {
         }
       );
     } else {
-      this.dialog.openDialog('Please log in!');
-    }
-  }
+      this.dialogConfig.data = {
+        description: 'Please log in!'
+      };
 
-  isUserLoggedIn(): any {
-    this.authService.isUserTokenValid()
-      .then(result => {
-        if (result) {
-          this.isLoggedIn = true;
-        } else {
-          this.isLoggedIn = false;
-        }
-      });
+      this.dialogService.openDialog(ErrorDialogComponent, this.dialogConfig);
+    }
   }
 
   registerSolo(deed: Deed) {
-    if (!this.isLoggedIn) {
-      this.dialog.openDialog('Please log in!');
+    if (!this.userService.loggedInStatus) {
+      this.dialogConfig.data = {
```

Comment:
couldn't you have implemented open dialog, so that you could pass data directly into openDialog() method? Maybe like this? openDialog(ErrorDialogComponent, {description: 'bla bla bla'})